### DISCUSSION TITLE
getting "SignatureDoesNotMatch" error while using AWS SNS.

### DISCUSSION THREAD

**Discussion by vibuverma** [jertel/elastalert2#1001] [Q&A] [answered]
while trying to get alerts on AWS SNS using the below-mentioned config, getting the "SignatureDoesNotMatch" error.

config:
```
alert: "sns"
sns_topic_arn: "arn:aws:sns:ap-south-1:514748507887:DemoTopic"
sns_aws_access_key_id: "XXX"
sns_aws_secret_access_key: "YYYY"
sns_aws_region: "ap-south-1"
```

error:
`ERROR:elastalert:Error while running alert sns: Error sending Amazon SNS: An error occurred (SignatureDoesNotMatch) when calling the Publish operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.`

**Comment by jertel** [comment, marked as answer]
Is this the first time you've attempted to use this IAM account programmatically (via CLI or SDK)?

**Reply by vibuverma** [reply]
Nope. Used this account for terraform as well.

**Reply by jertel** [reply]
Any time I have seen this error it was due to using the wrong AWS credentials, the AWS access key needing to be regenerated, or the account not setup for CLI access. From your above response you can eliminate that last one, so I suggest regenerating the access key/secret and trying again.

**Reply by nsano-rururu** [reply]
@jertel  @vibuverma 
There was a mistake when I added error handling before. I'm sorry. I will make a pull request from now on.

https://github.com/jertel/elastalert2/blob/master/elastalert/alerters/sns.py#L30

current
```
aws_secret_access_key=self.sns_aws_access_key_id,
```
fix
```
aws_secret_access_key=self.sns_aws_secret_access_key,
```

**Reply by nsano-rururu** [reply]
@jertel 
I made a pull request, so please give me a review.
https://github.com/jertel/elastalert2/pull/1003

**Reply by jertel** [reply]
Good find. Thanks for looking into that @nsano-rururu!
